### ISSUE TITLE
bool - holter tower

### ISSUE THREAD

**Issue by vegarringdal** [tomvandig/web-ifc#53]
Holter tower is missing a few windows
Tried this with v 0.20 with and without USE_FAST_BOOLS

![image](https://user-images.githubusercontent.com/2901416/122945308-59e37500-d32d-11eb-8a17-8b2ee0ca0d7a.png)

**Comment by agviegas** [comment]
Hey @vegarringdal can you confirm that this still happens? Cheers!

**Comment by vegarringdal** [comment]
@agviegas 

Nope, but noticed npm version and console is printing out different version...
I just cloned... so I dint not have any old files laying around

![image](https://user-images.githubusercontent.com/2901416/173848953-7ef444fd-18ae-4045-a544-00cb8567ad2a.png)

**Comment by agviegas** [comment]
Hey @vegarringdal yeah that's actually a mistake on our side, you are probably using 35, the log has a typo. If you could provide me the model privately, we'll address this asap. Cheers!

**Comment by elalish** [comment]
I'd love to know more about what's going on here. Are you referring to merging nearby verts to e.g. make an STL manifold? Like most things, the simple cases are simple, but making it robust can be pretty tricky. Certainly pass any `Manifold` errors back to our library. In fact, I have some flags for verbosity that help with spitting out polygon error cases (which is how I've created a lot of my tests). 

Speaking of, I had no idea you took me on as a dependency; that's exciting! Ideally I'd like to keep a list somehow of what projects are using the library, though it seems easier said than done. Anyway, thanks for mentioning me.

**Comment by makc** [comment]
@elalish I could log input mesh if you are willing to take a look at that in the context of polygon.cpp error;
> Are you referring to merging nearby verts to e.g. make an STL manifold

currently ifc input data is just a triangle soup - so I have attempted to merge the vertices back and when we have a simple enough mesh it works. but I guess there are cases where too many vertices touch and then merging them is doing nothing but harm.

**Comment by elalish** [comment]
Yeah, this is why triangle soup like STL is such a problem - it's basically impossible to reliably retain manifoldness in general cases. This is why I helped develop the 3MF format, which is focused on defining and retaining manifoldness. 

Yes, please log the input meshes you're having trouble with. If you can manage to write a PR with a `TEST` case that reproduces it, that would be the best, but an issue with some repro steps would also be helpful.

**Comment by makc** [comment]
@agviegas here is an example of non-manifold hole that breaks this tower file:
![image](https://user-images.githubusercontent.com/242577/175666021-8cc4aaab-3ec8-4076-b98a-d7a2d8d16e15.png)

again, 2 boxes together in one hole, just like in #126 - so I guess the same issue.

**Comment by agviegas** [comment]
Hey @elalish sorry for not letting you know, my bad. We will probably release more bounties to identify these errors, fix the ones related to geometry generation on our side and support your fantastic work (like we did with Garret's three-mesh-bvh), if that's ok with you. 🙂 We'll let you know of any news. In the meantime, of course, feel free to include IFC.js as one of the projects using your library as a dependency, and let us know if you need anything from us.

**Comment by elalish** [comment]
Fantastic, keep those bug reports coming! And certainly let me know if have any comments on the API.

**Comment by beachtom** [comment]
From what I can see this is now rendering correctly. So I will close this
